glib/src/object.rs

```diff
@@ -1726,42 +1729,17 @@ impl<T: ObjectType> ObjectExt for T {
         F: Fn(&[Value]) -> Option<Value>,
     {
         let signal_name: &str = signal_name.into();
-
         let type_ = self.get_type();
 
-        let mut signal_id = 0;
-        let mut signal_detail = 0;
+        let (signal_id, signal_detail) = SignalId::parse_name(signal_name, type_, true)
+            .ok_or_else(|| bool_error!("Signal '{}' of type '{}' not found", signal_name, type_))?;
+        let signal_query = signal_id.query();
 
-        let found: bool = from_glib(gobject_ffi::g_signal_parse_name(
-            signal_name.to_glib_none().0,
-            type_.to_glib(),
-            &mut signal_id,
-            &mut signal_detail,
-            true.to_glib(),
-        ));
-
-        if !found {
-            return Err(bool_error!(
-                "Signal '{}' of type '{}' not found",
-                signal_name,
-                type_
-            ));
-        }
-
-        let mut details = mem::MaybeUninit::zeroed();
-        gobject_ffi::g_signal_query(signal_id, details.as_mut_ptr());
-        let details = details.assume_init();
-        if details.signal_id != signal_id {
-            return Err(bool_error!(
-                "Signal '{}' of type '{}' not found",
-                signal_name,
-                type_
-            ));
+        if !signal_query.static_scope() {
```

Comment:
Does it make sense to have such validation here? I'm not very sure.

Reply:
No, that shouldn't be checked there